### PR TITLE
Posts controller

### PR DESCRIPTION
**Pull request by guragt**
1. Create window for post:
Screenshot 2020-10-29 at 14 47 25
2. Home page:
Screenshot 2020-11-03 at 17 33 42
3. Profile page:
Screenshot 2020-11-02 at 15 51 12
4. Edit page:
Screenshot 2020-11-03 at 17 37 19

### CODE REVIEW

File: app/views/posts/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<h2><%= t('.title') %></h2>
```
Comment (Snick555) [inline app/views/posts/new.html.erb]: make this form in a modal window

File: spec/controllers/posts_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let!(:user) { create(:user) }
+  let!(:post_valid_params) { attributes_for(:post)  }
+  let!(:post_invalid_params) { { description: 'a' * 2001, image: '' } }
+
```
Comment (Snick555) [inline spec/controllers/posts_controller_spec.rb:7]: make two contexts: 'unauthorized user' and 'authorized user' which will be common for all actions

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+
```
Comment (Snick555) [inline app/controllers/posts_controller.rb:5]: add index action and redirect to the index after the post was created

Comment (guragt) [reply app/controllers/posts_controller.rb:5]: Can i redirect to user profile page (action show in users controller), where all user's posts will be shown?

Comment (Snick555) [reply app/controllers/posts_controller.rb:5]: I think better will be index action here (all posts include user's posts)

File: spec/controllers/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      it { is_expected.to render_template('new') }
+    end
+
+    describe 'POST#create' do  
+      context 'with valid params' do
+        it 'create a new post' do
+          expect do
+            post :create, params: { post: post_valid_params }
+          end.to change(Post, :count).by(1)
+          expect(response).to have_http_status(:redirect)
```
Comment (Snick555) [inline spec/controllers/posts_controller_spec.rb]: `expect(response).to redirect_to(posts_path)`

File: spec/controllers/posts_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          expect(flash[:success]).not_to be_present
+        end
+      end
+    end
+  end
+
+  context 'unauthorized user' do
+    describe 'GET#new' do
+      subject { get :new }
+      it { is_expected.to have_http_status(:redirect) }
```
Comment (Snick555) [inline spec/controllers/posts_controller_spec.rb]: `expect(response).to redirect_to(login_path)`

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -4,6 +4,11 @@
     <nav>
       <ul class="nav navbar-nav navbar-right">
         <% if signed_in? %>
+          <li>
+            <button class="header-button" data-toggle="modal" data-target="#new-modal" } do %>
```
Comment (Snick555) [inline app/views/layouts/_header.html.erb]: typo in the end

File: app/controllers/posts_controller.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: :index
+
+  def index
+    redirect_to new_user_session_path unless signed_in?
```
Comment (Snick555) [inline app/controllers/posts_controller.rb]: here should be shown all posts, or it will be in next pr?

Comment (guragt) [reply app/controllers/posts_controller.rb]: in progress...

File: app/controllers/posts_controller.rb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: :index
+  before_action :obtain_post, only: %i[show]
 
   def index
     if signed_in?
```
Comment (Snick555) [inline app/controllers/posts_controller.rb]:
```
return redirect_to new_user_session_path if signed_in?

@posts = current_user.posts.order(created_at: :desc)
```

File: app/controllers/posts_controller.rb [hunk 9]
```diff
@@ -1,5 +1,6 @@
 class PostsController < ApplicationController
   before_action :authenticate_user!, except: :index
+  before_action :obtain_post, only: %i[show]
```
Comment (Snick555) [inline app/controllers/posts_controller.rb]: do we need it if we use it only in one action? but maybe in future perspective let it be

File: app/views/posts/_post.html.erb [hunk 10]
```diff
@@ -2,9 +2,20 @@
   <div class="post-header">
     <%=link_to image_tag(post.user.avatar.small_thumb.url), user_path(post.user), class: 'post-avatar'%>
     <%= link_to "#{post.user.nickname}", user_path(post.user), class: 'post-nick' %>
+    <% if current_user == post.user %>
+      <div class="dropdown post-dropdown">
+        <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+          <i class="glyphicon glyphicon-option-horizontal"></i>
+        </a>
+        <ul class="dropdown-menu">
+          <li><%= link_to t('.change'), "#" %></li>
```
Comment (Snick555) [inline app/views/posts/_post.html.erb]: `edit` sounds better than `change`

**Review by Snick555 — APPROVED** [review]

**Pull request merged**